### ISSUE TITLE
check_notebook_jit_surfaces names jax.grad and jax.value_and_grad in its fix message but scans neither

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#2063]
`check_notebook_jit_surfaces` tells the reader it scans `jax.grad` and `jax.value_and_grad`. It scans neither.

## What happens

The guard's failure message ends:

```
Fix: use only predict_photometry, predict_spectrum, or predict_properties
inside jax.jit / jax.vmap / jax.grad / jax.value_and_grad spans.
```

Four transforms named. `_find_jax_transforms` matches two:

```python
patterns = [
    r"\bjax\.jit\s*\(",
    r"\bjax\.vmap\s*\(",
    r"\bjit\s*\(",
    r"\bvmap\s*\(",
]
```

Measured:

```python
>>> _find_jax_transforms("jax.grad(lambda p: m.predict(p))")
[]
>>> _find_jax_transforms("jax.value_and_grad(lambda p: m.predict(p))")
[]
>>> _find_jax_transforms("grad(m.predict)")
[]
```

A notebook cell reading `jax.grad(lambda p: jnp.sum(model.predict(p).photometry()))`
passes the guard and fails at trace time — the #1255 class the guard exists
to catch. The module docstring is accurate about the two patterns; only the
advice the user is shown overreaches.

## Why it matters more than the count suggests

`jax.grad` over a likelihood is the single most common thing a fitting
notebook does, and it is the transform most likely to be written *without*
an enclosing `jax.jit` during exploration. The rich `Prediction` /
`PropertyCatalog` objects are deliberately not pytrees (NAMING_CONTRACT
§4b), so `grad` traces them and raises exactly as `jit` does. Coverage
stops precisely where the idiom is most common.

The gap survived because nothing asserted it either way. It is now pinned
in `tests/unit/test_check_notebook_jit_surfaces.py::test_gradient_transforms_are_not_scanned`,
which states the current behaviour so it stays visible.

## Fix

Add `jax.grad`, `jax.value_and_grad`, and their bare forms to `patterns`.

Two things to check when doing it, neither of which I have done here —
which is why this is filed rather than fixed alongside the coordinate fix
in the same PR:

1. **It changes what the live tree rejects.** `tools/check_notebook_jit_surfaces.py`
   currently reports `OK -- 64 notebook(s) clean.` Widening the patterns may
   turn some of those red, and each hit needs judging on its merits rather
   than being swept up in an unrelated change.
2. **Bare `grad(` is riskier than bare `jit(`.** `grad` is a common local
   variable name in gradient-descent code (`grad = jax.grad(f)(x)` then
   `grad(...)` would not match, but `grad` used as a function elsewhere
   might). The bare form may need to be narrower than the qualified one.

## Labels

area:ci, area:notebooks, bug, silent-failure
